File: original_scripts/julia/binary_search/deepseek-coder_6.7b.jl
```julia
function binary_search(arr::Array{T}, target::T) where T <: Number
    low = 1
    high = length(arr)
    
    while low <= high
        mid = (low + high) ÷ 2
        
        if arr[mid] == target
            return true
        elseif arr[mid] < target
            low = mid + 1
        else
            high = mid - 1
        end
    end
    
    return false
end
```

File: original_scripts/java/binary_search/deepseek-coder_6.7b.java
```java
public class BinarySearch {
    int binarySearch(int arr[], int low, int high, int x) {
        if (high >= low) {
            int mid = low + (high - low) / 2;
  
            // If the element is present at the middle itself
            if (arr[mid] == x) return mid;
  
            // If element is smaller than mid, then it can only be present in left subarray
            if (arr[mid] > x) return binarySearch(arr, low, mid - 1, x);
  
            // Else the element can only be present in right subarray
            return binarySearch(arr, mid + 1, high, x);
        }
  
        // We reach here when element is not present in array
        return -1;
    }
}
```